spring-cloud-azure-config/src/main/java/com/microsoft/azure/spring/cloud/config/msi/AzureInstanceMetadataService.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See LICENSE in the project root for
+ * license information.
+ */
+package com.microsoft.azure.spring.cloud.config.msi;
+
+import org.springframework.util.Assert;
+import retrofit2.http.Url;
+
+import java.io.UnsupportedEncodingException;
+import java.net.MalformedURLException;
+import java.net.URISyntaxException;
+import java.net.URL;
+import java.net.URLEncoder;
+
+/**
+ * Wraps functions provided by Azure Instance Metadata Service(IMDS)
+ */
+public class AzureInstanceMetadataService {
+    private String apiVersion = "2018-02-01-preview";
+    private String tokenEndpoint = "http://169.254.169.254/metadata/identity/oauth2/token";
+    private String objectId;
```

Comment:
What's the meaning of `objectId` and `clientId`? what's the impact of not providing them? Maybe some comments would help clarify this.

Reply:
It's tightly related with [Azure Instance Metadata Service](https://docs.microsoft.com/en-us/azure/active-directory/managed-identities-azure-resources/how-to-use-vm-token#get-a-token-using-http), and it's not invented by this code change, so comments might not be necessary here?

Reply:
Maybe put that link in the comment is enough.